src/components/AdvanceAnalyticsV2/tabs/Skills.jsx

```diff
@@ -21,7 +56,18 @@ const Skills = () => {
             description: 'Subtitle for the top skills chart.',
           })}
         />
-        <EmptyChart />
+        {(isLoading || isError) ? <ProgressOverlay isError={isError} /> : <ScatterChart
```

Comment:
Regardless, just keep in mind accessibility with regard to loading spinners. For example, rendering `Spinner` without `screenReaderText` does not inform non-sighted users using a screenreader (e.g., Voiceover) that something is loading. See the Paragon [props API](https://paragon-openedx.netlify.app/components/spinner/#props-api) for `Spinner`.

Reply:
This is how the UI looks like when we isLoading is true

![skills_tab_loading](https://github.com/user-attachments/assets/d61dda78-1abe-485e-b44c-32b8e065059c)

I tried using Suspense with react-query, but I ran into a challenge. Instead of having individual loaders for each component, I had to specify a single fallback loader, which covers all components. I'm unsure if this approach is ideal from a UX perspective.

Here’s what I’m considering:

* Initially, display individual loaders as shown in the above screenshot.
* If the charts have already been rendered and the user makes changes in the dropdowns, avoid showing the full-screen loaders as before. Instead, display small loaders beside the chart titles. This way, the stale charts will remain visible while they are updated in the background.